shim/src/handle_vm_kvm_create_irqchip.c

```diff
@@ -48,6 +49,13 @@ handle_vm_kvm_create_irqchip(struct shim_vm_t *const pmut_vm) NOEXCEPT
         bferror("The shim is not running in a VM. Did you forget to start MicroV?");
         return SHIM_FAILURE;
     }
+
+    platform_mutex_lock(&pmut_vm->mutex);
+    if (pmut_vm->is_irqchip_created) {
+        bferror("The IRQCHIP is not created, Did you forget to create it?");
+        return SHIM_FAILURE;
+    }
+
```

Comment:
An `irqchip` can only be created when no VCPU have been created yet. We should add a check for this here as well.

Reply:
Okay, here i should follow like the way it is done for irqchip check by adding a bool variable?

Reply:
The `shim_vm_t` has a `vcpus` member that you can iterate over to check if an fd is non-zero. Or you could add a `num_vcpus` field and modify `handle_vm_kvm_{create,destroy}_vcpu.c` to update it.

Reply:
This change is addressed